policy/handler.go

```diff
@@ -218,3 +278,56 @@ func (h *Handler) updateHandler(current, next *Policy) {
 		h.ticker = time.NewTicker(next.EvaluationInterval)
 	}
 }
+
+// enforceCooldown blocks until the cooldown period has been reached, or the
+// handler has been instructed to exit. When this function returns the caller
+// should perform a check to identify if the handler should exit or not. This
+// can be done:
+//
+// if ctx.Err() != nil || !h.running {
+//   return
+// }
+func (h *Handler) enforceCooldown(ctx context.Context, t time.Duration) {
+
+	// Log that cooldown is being enforced. This is very useful as cooldown
+	// blocks the ticker making this the only indication of cooldown to
+	// operators.
+	h.log.Debug("scaling policy has been placed into cooldown", "cooldown", t)
+
+	// Using a timer directly is mentioned to be more efficient than
+	// time.After() as long as we ensure to call Stop(). So setup a timer for
+	// use and defer the stop.
+	timer := time.NewTimer(t)
+	defer timer.Stop()
+
+	// Cooldown should not mean we miss other handler control signals. So wait
+	// on all the channels desired here.
+	for {
```

Comment:
This loop doesn't seem necessary. Once a channel is triggered we exit right away.